frontend/index.js

```diff
@@ -492,10 +493,34 @@ function getElementIds(list) {
   return list[ELEM_IDS]
 }
 
+/**
+ * Finds the latest integer index of a cursor object.
+ * If the character was deleted, returns -1.
+ * 
+ * todos:
+ * - consider returning the closest character if deleted
+ * - consider optimizing the linear search
+ */
+function findCursorIndex (cursor, doc, findClosest) {
```

Comment:
I think it's odd that this function requires `doc` as a separate argument, but I couldn't find any way to navigate from an object within a doc to the root doc object.